test/agent.js

```diff
@@ -873,8 +871,6 @@ test('#captureError()', function (t) {
         this.agent.captureError(new Error('with callback'), function () {
           t.pass('called callback')
         })
-        span.end()
-        trans.end()
```

Comment:
Maybe put this in a `t.on('end', () => { ... })` just to ensure they are still ended before the test exits.

Reply:
What's your concern about it? We start transactions/spans in other tests that we never end either, so if there's a problem with this, we should probably look at this in general.

Reply:
Mainly just avoiding transactions leaking into other tests and causing strange behaviour. I suppose I'm fine landing this as-is though.